Description: don't crash on corrupted svg
Author: Bruno Girin <brunogirin@gmail.com>
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=626559
Bug-Ubuntu: https://bugs.launchpad.net/bugs/608026

=== modified file 'rsvg-paint-server.c'
Index: librsvg-2.32.1/rsvg-paint-server.c
===================================================================
--- librsvg-2.32.1.orig/rsvg-paint-server.c	2010-05-31 07:21:34.000000000 +1000
+++ librsvg-2.32.1/rsvg-paint-server.c	2010-11-16 09:30:22.813990003 +1100
@@ -118,7 +118,7 @@
     guint32 rgb;
     if (inherit != NULL)
         *inherit = 1;
-    if (!strcmp (str, "none"))
+    if (str == NULL || !strcmp (str, "none"))
         return NULL;
 
     name = rsvg_get_url_string (str);
