From dfe34c8757debd07d4ef2f6f0381b2bcac1addc0 Mon Sep 17 00:00:00 2001
From: Ting-Wei Lan <lantw44@gmail.com>
Date: Mon, 7 Aug 2017 21:43:10 -0500
Subject: [PATCH] bgo#782098 - Don't pass deprecated --nogtkinit to
 gtkdoc-scangobj

---
 doc/Makefile.am | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/doc/Makefile.am b/doc/Makefile.am
index 5db7716..5decdda 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -23,7 +23,7 @@ DOC_MAIN_SGML_FILE = $(DOC_MODULE)-docs.xml
 DOC_SOURCE_DIR = ..
 
 # Extra options to pass to gtkdoc-scangobj. Not normally needed.
-SCANGOBJ_OPTIONS = --nogtkinit --type-init-func="g_type_init ()"
+SCANGOBJ_OPTIONS = --type-init-func="g_type_init ()"
 
 # Extra options to supply to gtkdoc-scan.
 # e.g. SCAN_OPTIONS=--deprecated-guards="GTK_DISABLE_DEPRECATED"
@@ -96,7 +96,7 @@ HTML_IMAGES =
 # Extra SGML files that are included by $(DOC_MAIN_SGML_FILE).
 # e.g. content_files=running.sgml building.sgml changes-2.0.sgml
 content_files = \
- 	version.xml
+	version.xml
 
 # SGML files where gtk-doc abbrevations (#GtkWidget) are expanded
 # These files must be listed here *and* in content_files
-- 
2.15.0.rc1

