Index: librsvg-2.40.13/rsvg-convert.c
===================================================================
--- librsvg-2.40.13.orig/rsvg-convert.c	2016-01-29 14:16:21.806607812 +0100
+++ librsvg-2.40.13/rsvg-convert.c	2016-01-29 14:17:33.448096327 +0100
@@ -140,6 +140,8 @@
     RsvgDimensionData dimensions;
     FILE *output_file = stdout;
     char *export_lookup_id;
+    char *basename;
+    gboolean rsvg_compat = FALSE;
 
 #ifdef G_OS_WIN32
     HANDLE handle;
@@ -199,6 +201,27 @@
         return 0;
     }
 
+    if (args)
+        while (args[n_args] != NULL)
+            n_args++;
+
+    basename = g_path_get_basename (argv[0]);
+    if (g_ascii_strcasecmp (basename, "rsvg") == 0) {
+        rsvg_compat = TRUE;
+    }
+    g_free (basename);
+
+    if (rsvg_compat) {
+        /* We are being called as rsvg. Apply some compat magic */
+        if (n_args == 2) {
+            output = g_strdup (args[1]);
+            n_args = 1;
+        } else {
+            fprintf (stderr, _("Please supply an input and output file.\n"));
+            exit (1);
+        }
+    }
+
     if (output != NULL) {
         output_file = fopen (output, "wb");
         if (!output_file) {
@@ -210,10 +233,6 @@
         g_free (output);
     }
 
-    if (args)
-        while (args[n_args] != NULL)
-            n_args++;
-
     if (n_args == 0) {
         n_args = 1;
         using_stdin = TRUE;
