--- rsvg-file-util.c.old	2005-06-30 22:56:32.791188192 +0200
+++ rsvg-file-util.c	2005-06-30 22:58:10.219376856 +0200
@@ -81,7 +81,13 @@
 		
 		zoomx = (double) real_data->width / *width;
 		zoomy = (double) real_data->height / *height;
-		zoom = MIN (zoomx, zoomy);
+
+		if (zoomx < 0)
+			zoom = zoomy;
+		else if (zoomy < 0)
+			zoom = zoomx;
+		else
+			zoom = MIN (zoomx, zoomy);
 		
 		*width = floor (zoom * *width + 0.5);
 		*height = floor (zoom * *height + 0.5);
--- test-rsvg.c.old	2005-06-30 22:48:24.234460104 +0200
+++ test-rsvg.c	2005-06-30 22:59:14.916541392 +0200
@@ -102,7 +102,14 @@
 		pixbuf = rsvg_pixbuf_from_file_at_zoom (args[0], x_zoom, y_zoom, NULL);
 	/* if both are unspecified, assume user wants to resize pixbuf in at least 1 dimension */
 	else if (x_zoom == 1.0 && y_zoom == 1.0)
-		pixbuf = rsvg_pixbuf_from_file_at_size (args[0], width, height, NULL);
+		{
+			/* if one parameter is unspecified, assume user
+			 * wants to keep the aspect ratio */
+			if (width == -1 || height == -1)
+				pixbuf = rsvg_pixbuf_from_file_at_max_size (args[0], width, height, NULL);
+			else
+				pixbuf = rsvg_pixbuf_from_file_at_size (args[0], width, height, NULL);
+		}
 	else
 		/* assume the user wants to zoom the pixbuf, but cap the maximum size */
 		pixbuf = rsvg_pixbuf_from_file_at_zoom_with_max (args[0], x_zoom, y_zoom,
