--- rsvg-file-util.c.old	2005-06-30 19:54:08.430983840 +0200
+++ rsvg-file-util.c	2005-06-30 19:57:47.282713320 +0200
@@ -280,7 +280,11 @@
 	data.type = RSVG_SIZE_WH;
 	data.width = width;
 	data.height = height;
-	data.keep_aspect_ratio = FALSE;
+	if (width == -1 || height == -1) {
+	    data.keep_aspect_ratio = TRUE;
+	} else {
+	    data.keep_aspect_ratio = FALSE;
+	}
 	
 	return rsvg_pixbuf_from_file_with_size_data_ex (handle, file_name, &data, error);
 }
