--- rsvg-art-paint-server.c.orig	2005-10-08 05:29:33.000000000 +0200
+++ rsvg-art-paint-server.c	2005-10-12 00:05:23.000000000 +0200
@@ -228,6 +228,9 @@
 		scale = 100000000.;
 	else
 		scale = 1.0 / (dx * dx + dy * dy);
+	if (isnan(scale)) scale = 0.;
+	if (isnan(dx)) dx = 0.;
+	if (isnan(dy)) dy = 0.;
 	agl->a = dx * scale + FUDGE;
 	agl->b = dy * scale + FUDGE;
 	agl->c = -(x1 * agl->a + y1 * agl->b) + FUDGE;
